test/arrayops.jl

```diff
@@ -362,6 +362,14 @@ a = [0,1,2,3,0,1,2,3]
 @test findprev(isodd, [2,4,5,3,9,2,0], 7) == 5
 @test findprev(isodd, [2,4,5,3,9,2,0], 2) == 0
 
+# find with general iterables
+s = "julia"
+@test find(s) == [1,2,3,4,5]
```

Comment:
I didn't have a use case in mind for `find("string")`; I was primarily interested in opening up `find(testf::Function, A)` for strings. Returning all indices for a string makes sense to me though, at least more so than returning indices of non-null bytes. I suppose that isn't *so* bad, just kind of weird.